File: 2390-removing-stars-from-a-string/2390-removing-stars-from-a-string.py
class Solution:
    def removeStars(self, s: str) -> str:

        stack = []

        for val in s:
            #While stack is not empty 
            while stack and (stack[-1].isalpha() and val == "*"):
                stack.pop()
                break
            else:
                stack.append(val)
        
        return "".join(stack)

        

File: 2390-removing-stars-from-a-string/2390-removing-stars-from-a-string.cpp
class Solution {
public:
    string removeStars(string s) 
    {
        int n = s.size();
        std::stack<char> stack;

        for (int i{}; i < n; i++)
        {
            //Grab char
            char c = s[i];

            if (c == '*')
            {
                char x = stack.top();
                stack.pop();
            }
            else
            {
                stack.push(c);
            }
        }        

        while (!stack.empty())
        {
            res += stack.top();
            stack.pop();
        }   

        reverse(res.begin(), res.end());

        return res;
    }

private:
    std::string res = "";
};